GLOSSARY.md

```diff
@@ -0,0 +1,31 @@
+# Glossary
+
+This document gathers the canonical nomenclature of the Lightning concepts used by the LDK contributors. Note, they may diverge from the BOLT specs or the ones used by the wider Lightning community, the test of time has revealed that it can be hard to agree on terms employed. The following terms are employed with a best effort across the codebase aiming to reduce both code reviewers and library users confusion.
+
+### Channel Initiator
+
+In LDK context, the channel initiator is the entity who is taking the decision to open a channel. Finalization is relying on counterparty channel acceptance policy. Within current protocol version, the initiator has the burden to pay channel onchain fees.
+
+### Holder/Counterparty
+
+Inspired by financial cryptography, the holder is the entity beneficiary of a right. In LDK context, the holder is the entity who is served by the implementation in its channel operations. The counterparty is the entity participating in the channel operation as a peer, but of whom the interests are not of concern by the implementation.
+
+Used across the channel-management subsystem (`Channel`).
+
+### Broadcaster/Countersignatory
+
+In Lightning, states are symmetric but punishment is asymmetric, which forces channel parties to hold different commitment transactions. At transaction construction, the broadcaster designates the entity which has the unilateral capability to broadcast the tranaction. The countersignatory is the entity providing signatures for the broadcastable transaction and thus verifying its encodes the off-chain states. At any point in time, there should be two "latest" commitment transactions that have been processed by each party's implementation, one where the holder is the broadcaster and counterparty is countersignatory, and one where the holder is the countersignatory and the counterparty is the broadcaster.
+
+Used across the channel-utils library (`chan_utils.rs`).
+
+### Watchtower
+
+An external untrusted service, which can only publish justice transactions. The security property of a watchtower is that if you subscribe to n of them, you will be safe if at least 1 does the right thing.
```

Comment:
s/, which/that

Consider making a complete sentence as it makes it clear why there is a distinction between the two phrasings:

> A watchtower is an external untrusted service that can only publish justice transactions.